android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/views/HomeSurveysView.kt

```diff
@@ -0,0 +1,223 @@
+package co.nimblehq.ic.kmm.suv.android.ui.screens.home.views
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import co.nimblehq.ic.kmm.suv.android.R
+import co.nimblehq.ic.kmm.suv.android.extension.placeholder
+import co.nimblehq.ic.kmm.suv.android.ui.components.DotsIndicator
+import co.nimblehq.ic.kmm.suv.android.ui.theme.AppTheme
+import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
+import co.nimblehq.ic.kmm.suv.android.util.LoadingParameterProvider
+import coil.compose.AsyncImage
+import timber.log.Timber
+
+data class HomeSurveyUiModel(
+    val title: String,
+    val description: String,
+    val imageUrl: String
+)
+
+data class HomeSurveysUiModel(
+    val surveys: List<HomeSurveyUiModel>,
+    val currentPageIndex: Int,
+    val isLoading: Boolean
+) {
+    val totalPages: Int = surveys.size
```

Comment:
we should use `get() =` here as `surveys.size` could be changed after initializing

```suggestion
    val totalPages: Int
        get() = surveys.size
```

Reply:
@luongvo So I guess I should change from `val` to `var`, right?
```kotlin
data class HomeSurveysUiModel(
    var surveys: List<HomeSurveyUiModel>,
    var currentPageIndex: Int,
    var isLoading: Boolean
)
```

Just curious, normally, we will create a new UiModel or update the current UiModel? 🤔

Reply:
@suho 

- `surveys.size` could be changed after initializing. As it's a List, we can assign MutableList into List data type so the size could be changed.
- We don't need to change all class fields to `var`. What I meant is we should use
```
    val totalPages: Int
        get() = surveys.size
```
instead of
```
    val totalPages: Int = surveys.size
```

Check this out for more detail: https://kotlinlang.org/docs/properties.html#getters-and-setters

Reply:
@luongvo Noted about `MutableList`. I did not think about it 😆

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/106/commits/029cfc14c4b123b651f9b3115660969719838097